question18/main.js

```diff
@@ -0,0 +1,191 @@
+} 
+
+const getData = new Promise((resolve) => {
+  showLoadImg();
+  setTimeout(() => {
+    resolve(fetch('https://mocki.io/v1/09593e78-80ba-4765-940e-3c70fb46078c'));
+  }, 3000);
+})
+
+//Get json data
+const callApi = async() => {
+  try {
+    const response = await getData;
+    if (!response.ok) {
+      console.error(`${response.status}:${response.statusText}`);
+    }
+    const json = await response.json();
+    return json.data;
+  }
+  catch(error) {
+    console.error(error);
+  }
+  finally {
+    removeLoadImg();
+  }
+}
+
+const init = async() => {
+  const imgData = await callApi();
+  if (imgData){
+    renderImgUiElement(imgData);
+    } else {
+    imgListsWrapper.textContent = "データの表示に失敗しました";
+    console.error('データの表示に失敗しました'); 
+  }
+}
+init();
+
+
+const renderImgUiElement = (imgData) => {
+  renderNextBtn(imgData);
+  renderPrevBtn(imgData);
+  renderPageNumElement(imgData);
+  renderPagenation(imgData);
+  clickedPagenation(imgData);
+  autoImgSwitch(imgData);
+  imgLists.appendChild(getImgListsFragment(imgData));
+}
+
+const renderPageNumElement = (imgData) => {
+  const numberOfPage = createElementWithClassName("p", "pageNumber");
+  imgListsWrapper.appendChild(numberOfPage);
+  numberOfPage.id = "js-number";
+  numberOfPage.textContent = `1 / ${imgData.length}`;  
+}
+
+
+const getImgListsFragment = (imgData) => {
+  const fragmentImglists = document.createDocumentFragment();
+  for (let i = 0; i < imgData.length; i++) {
+    const imgList = createElementWithClassName("li", "imgList");
+    const img = createElementWithClassName("img", `img_0${[i + 1]}`);
+    imgList.dataset.index = i;
+    img.src = imgData[i].img;
+    img.alt = imgData[i].description;
+    i === 0 && imgList.classList.add("is-show");
+    fragmentImglists.appendChild(imgList).appendChild(img);
+  }
+  return fragmentImglists;
+};
+
+
+let imgNum = 0;
+const imgList = document.getElementsByClassName("imgList");
+
+const renderPrevBtn = (imgData) => {
+  const prevBtn = createElementWithClassName("button", "prev");
+  prevBtn.id = 'js-prevbtn';
+  prevBtn.textContent = '◀︎';
+  imgListsWrapper.appendChild(prevBtn);
+  prevBtn.disabled = true;
+
+  prevBtn.addEventListener("click", function() {
+    imgNum -= 1;
+    document.querySelector(".is-show").classList.remove('is-show'); 
+    imgList[imgNum].classList.add('is-show');
+    switchPagenation(imgNum);
+    switchDisableForBtn(imgData); 
+    setNumberOfPage(imgData);
+  });
+}
+
+const renderNextBtn = (imgData) => {
+  const nxtBtn = createElementWithClassName("button", "next");
+  nxtBtn.id = 'js-nextbtn';
+  nxtBtn.textContent = '▶︎';
+  imgListsWrapper.appendChild(nxtBtn);
+
+  nxtBtn.addEventListener("click", function() {
+    imgNum += 1;
+    document.querySelector(".is-show").classList.remove('is-show'); 
+    imgList[imgNum].classList.add('is-show');
+    switchPagenation(imgNum);
+    switchDisableForBtn(imgData);
+    setNumberOfPage(imgData);
+  });
+}
```

Comment:
IMO
I feel that many of the processes are similar and could be commonized.
The image is like this.
Create an array showing the direction and create the elements in a loop.
```js
const ボタン作る関数 = (imgData) => {
  const directions = ["next", "prev"];

  directions.forEach((direction) => {
    const btn = createElementWithClassName("button", "prev");
    btn.id = `js-${direction}btn`;
    btn.textContent = direction === "prev" ? "◀︎" : "▶︎";
    imgListsWrapper.appendChild(btn);
    btn.disabled = direction === "prev";

    btn.addEventListener("click", function () {
      direction === "prev" ? (imgNum -= 1) : (imgNum += 1);
      document.querySelector(".is-show").classList.remove("is-show");
      imgList[imgNum].classList.add("is-show");
      switchPagenation(imgNum);
      switchDisableForBtn(imgData);
      setNumberOfPage(imgData);
    });
  });
};

```

Reply:
Thank you for your ideas!! :sob: I'm so glad I didn't realize you could put it together like this! I fixed it so please confirm.
fixed commit : https://github.com/csakyo/homework/pull/32/commits/187186d270fe31b91577974a3f18efd6831350dc